lib/Conversion/TTIRToD2M/TTIRToD2M.cpp

```diff
@@ -130,8 +148,25 @@ class D2MNamedRewriterCommon {
                               mlir::ConversionPatternRewriter &rewriter) const {
     if (isTTNNTensor(value.getType())) {
       assert(ttnnMode && "Unexpected TTNN tensor as op operand");
-      return rewriter.create<ttir::TTNNMetalLayoutCastOp>(
-          value.getLoc(), getMetalTensorFromTTNNTensor(rewriter, value), value);
+      auto metalTensorType = getMetalTensorFromTTNNTensor(rewriter, value);
+      auto metalCastOp = rewriter.create<ttir::TTNNMetalLayoutCastOp>(
+          value.getLoc(), metalTensorType, value);
+
+      ttcore::MemorySpace metalTensorMemSpace =
+          mlir::cast<ttcore::MetalLayoutAttr>(metalTensorType.getEncoding())
+              .getMemorySpace();
+      if (metalTensorMemSpace == ttcore::MemorySpace::DeviceL1) {
+        // Reblock L1 operand to unit grid to align with other operands while
+        // preserving original TTNN tensor shape. These views will be removed in
+        // GridSelection by insertTTNNDRAMStreams().
+        auto unitReblockingView = rewriter.create<d2m::ViewLayoutOp>(
+            value.getLoc(), reblockTensor(metalTensorType, {1, 1}),
+            metalCastOp->getResult(0));
+        return unitReblockingView.getResult();
+      } else {
+        // For DRAM operands, we can return the metal cast result directly.
```

Comment:
:warning: **llvm\-else\-after\-return** :warning:
do not use `` else `` after `` return ``
```suggestion
      }         // For DRAM operands, we can return the metal cast result directly.
```

Reply:
Fixed in latest version 👍